File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/MerchantMappingBatchConfirmItemResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import java.time.LocalDate;

import com.ssafy.e106.domain.subscription.entity.Subscription;

public record MerchantMappingBatchConfirmItemResponse(
    Long subscriptionId,
    Long serviceId,
    String serviceName,
    Long servicePlanId,
    Integer monthlyPrice,
    LocalDate nextBillingDate) {

  public static MerchantMappingBatchConfirmItemResponse from(Subscription subscription) {
    return new MerchantMappingBatchConfirmItemResponse(
        subscription.getSubscriptionId(),
        subscription.getService().getServiceId(),
        subscription.getService().getName(),
        subscription.getServicePlan().getServicePlanId(),
        subscription.getServicePlan().getMonthlyPrice(),
        subscription.getNextBillingDate());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/MerchantMappingBatchConfirmItemResponse.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class MerchantMappingBatchConfirmItemResponse(
    val subscriptionId: Long,
    val serviceId: Long,
    val serviceName: String,
    val servicePlanId: Long,
    val monthlyPrice: Int,
    val nextBillingDate: String? = null,
)
